watcher.go

```diff
@@ -94,6 +101,7 @@ func NewWatcher(i WatcherInput) *Watcher {
 		clients = NewClientSet()
 	}
 
+	bufferTriggerCh := make(chan string, dataBufferSize/2)
```

Comment:
Probably missed the logic in my skimming.. but I'm curious about why you used that buffer size (dataBufferSize/2) on the chanel?

Reply:
honestly it was arbitrary buffer size to reuse an existing variable while not as large of a buffer since the ratio of dependencies : templates end up very different. Lmk if you prefer another value @eikenb

Reply:
No, that's fine. Just was curious. Thanks.